testing/test-mathjax.tsx

```diff
@@ -23,11 +23,24 @@ type Props = {
 };
 
 export function TestMathjax({children: tex, onRender}: Props) {
-    const {domElement} = renderer.render(tex);
+    const ref = React.useRef<HTMLSpanElement>(null);
+    const {domElement, addLabelWhenPresentational} = React.useMemo(
+        () => renderer.render(tex),
+        [tex],
+    );
 
     React.useEffect(() => {
         renderer.updateStyles();
         onRender?.();
     }, [tex, onRender]);
-    return <span dangerouslySetInnerHTML={{__html: domElement.outerHTML}} />;
+
+    React.useLayoutEffect(() => {
+        if (ref.current) {
+            addLabelWhenPresentational(ref.current);
+            ref.current.innerHTML = "";
+            ref.current.appendChild(domElement);
+        }
+    });
```

Comment:
Minor: Since `useLayoutEffect` callbacks run before `useEffect` callbacks, I wonder if this should be swapped with the `useEffect` block above, so the code executes in the same order that you read it.

Reply:
swapped!